### ISSUE TITLE
Hound's HAML linter should support custom Ruby styles for RuboCop

### ISSUE THREAD

**Issue by gylaz** [houndci/hound#1121]
[HAML-Lint](https://github.com/brigade/haml-lint) supports RuboCop's configuration when `RuboCop` linter is enabled. Hound's implementation doesn't support custom Ruby styles. It should for parity.

**Comment by bensaufley** [comment]
+1 for this. This has been causing us numerous issues for a few months now.

**Comment by cguillemette** [comment]
+1

**Comment by bensaufley** [comment]
What's the word here? Until this is fixed, or _some method_ for specifying Ruby version for HAML is implemented at least, HAML linting straight-up won't work for a lot of people. More people all the time, as they upgrade their environments and take advantage of newer things like the safe navigation operator.

**Comment by gylaz** [comment]
We haven't made any progress on this, and I don't have an ETA but it's not high on our priorities right now.

I'd like to add that we advocate avoiding Ruby code in views (as much as possible), preferring helpers/presenters instead. Thus, for the most part RuboCop linting isn't as important as catching HAML syntax and style issues. I realize that Ruby still gets written in views -- heavier on some projects than others -- so it's still important to support RuboCop, and we'll support this eventually (perhaps even sooner than I think).

**Comment by bensaufley** [comment]
I agree that one should avoid filling one's views with Ruby, but as it stands, a single instance of `could_be_nil&.attribute` will break Hound for the view. It won't continue linting the file because it no longer understands the context.

**Comment by benreyn** [comment]
Seems related to #1499

**Comment by salbertson** [comment]
@gylaz is this already being done?

**Comment by gylaz** [comment]
I don't believe so.
